# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../tcp_wrappers/no-sys_errlist.patch
# Copyright (C) 2020 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- tcp_wrappers_7.6/percent_m.c.vanilla	2020-08-09 19:40:28.624502714 +0200
+++ tcp_wrappers_7.6/percent_m.c	2020-08-09 19:41:48.452501768 +0200
@@ -12,12 +12,6 @@
 #include <errno.h>
 #include <string.h>
 
-/*extern int errno;*/
-#ifndef SYS_ERRLIST_DEFINED
-extern char *sys_errlist[];
-extern int sys_nerr;
-#endif
-
 #include "mystdarg.h"
 
 char   *percent_m(obuf, ibuf)
@@ -29,10 +23,8 @@
 
     while (*bp = *cp)
 	if (*cp == '%' && cp[1] == 'm') {
-	    if (errno < sys_nerr && errno > 0) {
-		strcpy(bp, sys_errlist[errno]);
-	    } else {
-		sprintf(bp, "Unknown error %d", errno);
+	    if (errno) {
+		strcpy(bp, (strerror(errno) ? strerror(errno) : "Unknown error"));
 	    }
 	    bp += strlen(bp);
 	    cp += 2;
